src/lib.rs

```diff
@@ -3375,6 +3377,34 @@ where
     }
 }
 
+/// A wrapper struct for serializing errors references.
+///
+/// This struct can be used to wrap types that don't implement `slog::Value` but
+/// do implement `std::error::Error` so that they can be logged.
+/// This is usually not used directly but using `#error` in the macros.
+///
+/// This struct is only available in `std` because the `Error` trait is not available
+/// without `std`.
+///
+/// Use [`ErrorValue`] if you want to move ownership of the error value.
+#[cfg(feature = "std")]
+pub struct ErrorRef<'a, E: std::error::Error>(pub &'a E);
+
+#[cfg(feature = "std")]
+impl<'a, E> Value for ErrorRef<'a, E>
+where
+    E: 'static + std::error::Error,
+{
+    fn serialize(
+        &self,
+        _record: &Record<'_>,
+        key: Key,
+        serializer: &mut dyn Serializer,
+    ) -> Result {
+        serializer.emit_error(key, self.0)
```

Comment:
Maybe we should add an `emit_error_ref` method with a default implementation?

It is unfortunate we have the `'static` bound, but that's probably a requirement for things like `slog_async`. An alternative would be using `format_args!` and [`emit_arguments`], but that would loose information 😦.

[`emit_arguments`]: https://docs.rs/slog/2.7.0/slog/trait.Serializer.html#tymethod.emit_arguments

Reply:
That's a good point, may be worth adding it in the future